2996_reflection/reflection.md

```diff
@@ -271,4 +397,172 @@ std::string name1 = name_of(^S);             // Also okay.
 
 ## Metafunctions
 
+We propose a number of metafunctions declared in namespace `std::meta` to operator on reflection values.
+Adding metafunctions to an implementation is expected to be relatively "easy" compared to implementing the core language features described previously.
+However, despite offering a normal consteval C++ function interface, each on of these relies on "compiler magic" to a significant extent.
+
+### `invalid_reflection`, `is_invalid`, `diagnose_error`
+
+:::bq
+```c++
+consteval auto invalid_reflection(
+                  std::string_view message,
+                  std::source_location src_loc = std::source_location::current())->info;
+consteval auto is_invalid(info)->bool;
+consteval auto is_invalid(std::span<info>)->bool;
+consteval void diagnose_error(info);
+```
+:::
+
+An invalid reflection represents a potential diagnostic for an erroneous construct.
+Some standard metafunctions will generate such invalid reflections, but user programs can also create them with the `invalid_reflection` metafunction.
+`is_invalid` returns true if it is given an invalid reflection or a span containing at least one invalid reflection.
+Evaluating `diagnose_error` renders a program ill-formed.
+If the given reflection is for an invalid reflection, an implementation is encouraged to render the encapsulated message and source position as part of the diagnostic indicating that the program is ill-formed.
```

Comment:
Okay so like:
```cpp
std::meta::diagnose_error(my_info);
```
is basically like:
```cpp
if (std::meta::is_invalid(my_info)) {
    std::constexpr_error("Error: my_info is invalid reflection! msg={:?} location={}",
        /* msg? */,
        /* loc? */);
}
```
Right (this is with P2758's `std::constexpr_error`)?